components/board/customlinux/config.go

```diff
@@ -0,0 +1,53 @@
+package customlinux
+
+import (
+	"os"
+
+	"go.viam.com/utils"
+
+	"go.viam.com/rdk/components/board/genericlinux"
+)
+
+// GenericLinuxPin describes a gpio pin on a linux board.
+type GenericLinuxPin struct {
+	Name            string `json:"name"`
+	Ngpio           int    `json:"ngpio"`
+	RelativeID      int    `json:"relative_id"`
+	PWMChipSysFSDir string `json:"pwm_chip_dir,omitempty"`
+	PWMID           string `json:"pwm_id,omitempty"`
+}
+
+// GenericLinuxPins describes a list of pins on a linux board.
+type GenericLinuxPins struct {
+	Pins []GenericLinuxPin `json:"pins"`
+}
+
+// Validate ensures all parts of the config are valid.
+func (config *GenericLinuxPin) Validate(path string) error {
+	if config.Name == "" {
+		return utils.NewConfigValidationFieldRequiredError(path, "name")
+	}
+
+	if config.PWMChipSysFSDir != "" && config.PWMID == "" {
+		return utils.NewConfigValidationFieldRequiredError(path, "pwm_id")
```

Comment:
Put in the opposite check, too, where the ID is set but the directory is not.

Reply:
this case doesn't raise an error when parsed- the pwm chip won't work until a directory is provided but not sure if we should enforce providing both

Reply:
The point of the `Validate` function is to check whether the config looks plausible. You could certainly skip this and get errors later about how we can't find a directory named `""`, but I suspect it would provide a better UX if here we said if you supply a PWM ID, you also need to name the chip in the config. but I'm open to disagreement

Reply:
if the directory is an empty string we won't look for it, nor will we look at the PWMID value.

Reply:
Oh, that makes sense! So then comes the question of whether specifying the line on the chip but not the chip should be considered an error, or merely something to be ignored. I'm inclined to call it an error, but could be swayed to another position.